Client/Frontend/UserContent/UserScripts/DeAMP.js

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"use strict";
+
+(_ => {
+  const W = window
+  const D = W.document
+
+  let timesToCheck = 5
+  let intervalId
+
+  const checkIfShouldStopChecking = _ => {
+    timesToCheck -= 1
+    if (timesToCheck === 0) {
+      W.clearInterval(intervalId)
+    }
+  }
+
+  const checkForAmp = _ => {
+    const htmlElm = document.documentElement
+    const headElm = document.head
+
+    if (!headElm && !htmlElm) {
+      // checked too early and page structure not available.
+      checkIfShouldStopChecking()
+      return
+    }
+
+    if (!htmlElm.hasAttribute('amp') && !htmlElm.hasAttribute('⚡')) {
+      // We know this isn't an amp document, and no point of checking further
+      W.clearInterval(intervalId)
+      return
+    }
+
+    const canonicalLinkElm = D.querySelector('head > link[rel="canonical"][href^="http"]')
+    if (!canonicalLinkElm === null) {
+      // didn't find a link elm.
+      checkIfShouldStopChecking()
+      return
+    }
+
+    const targetHref = canonicalLinkElm.getAttribute('href')
+    try {
+      const destUrl = new URL(targetHref)
+      if (W.location.href == destUrl.href) {
+        // Avoid an infinite loading loop in some cases
+        W.clearInterval(intervalId)
+        return
+      }
+      W.location.replace(destUrl.href)
+    } catch (_) {
+      // Invalid canonical URL detected
+      W.clearInterval(intervalId)
+      return
+    }
+  }
+
+  intervalId = W.setInterval(checkForAmp, 1000)
```

Comment:
my fault nit, but can this timeout be changed to 250, and `timesToCheck` be changed to 20?